gtk4/src/bitset_iter.rs

```diff
@@ -0,0 +1,167 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::Bitset;
+use glib::translate::*;
+
+#[derive(Copy, Clone)]
+pub struct BitsetIter(ffi::GtkBitsetIter);
+
+impl BitsetIter {
+    #[doc(alias = "gtk_bitset_iter_init_at")]
+    pub fn init_at(set: &Bitset, target: u32) -> Option<(Self, u32)> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut iter = BitsetIter::uninitialized();
+            let mut value = std::mem::MaybeUninit::uninit();
+            let found: bool = from_glib(ffi::gtk_bitset_iter_init_at(
+                iter.to_glib_none_mut().0,
+                set.to_glib_none().0,
+                target,
+                value.as_mut_ptr(),
+            ));
+            if found {
+                Some((iter, value.assume_init()))
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "gtk_bitset_iter_init_first")]
+    pub fn init_first(set: &Bitset) -> Option<(Self, u32)> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut iter = BitsetIter::uninitialized();
+            let mut value = std::mem::MaybeUninit::uninit();
+            let found: bool = from_glib(ffi::gtk_bitset_iter_init_first(
+                iter.to_glib_none_mut().0,
+                set.to_glib_none().0,
+                value.as_mut_ptr(),
+            ));
+            if found {
+                Some((iter, value.assume_init()))
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "gtk_bitset_iter_init_last")]
+    pub fn init_last(set: &Bitset) -> Option<(Self, u32)> {
+        assert_initialized_main_thread!();
+        unsafe {
+            let mut iter = BitsetIter::uninitialized();
+            let mut value = std::mem::MaybeUninit::uninit();
+            let found: bool = from_glib(ffi::gtk_bitset_iter_init_last(
+                iter.to_glib_none_mut().0,
+                set.to_glib_none().0,
+                value.as_mut_ptr(),
+            ));
+            if found {
+                Some((iter, value.assume_init()))
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "gtk_bitset_iter_get_value")]
+    pub fn value(&self) -> u32 {
+        unsafe { ffi::gtk_bitset_iter_get_value(self.to_glib_none().0) }
+    }
+
+    #[doc(alias = "gtk_bitset_iter_is_valid")]
+    pub fn is_valid(&self) -> bool {
+        unsafe { from_glib(ffi::gtk_bitset_iter_is_valid(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "gtk_bitset_iter_next")]
+    pub fn next(&mut self) -> Option<u32> {
```

Comment:
This looks like it should implement `Iterator`. But not `DoubleEndedIterator` because `previous()` seems to behave difference

Reply:
clippy doesn't like this still

```
error: method `next` can be confused for the standard trait method `std::iter::Iterator::next`
  --> gtk4/src/bitset_iter.rs:79:5
   |
79 | /     pub fn next(&mut self) -> Option<u32> {
80 | |         unsafe {
81 | |             let mut value = std::mem::MaybeUninit::uninit();
82 | |
...  |
92 | |         }
93 | |     }
   | |_____^
   |
   = note: `-D clippy::should-implement-trait` implied by `-D warnings`
   = help: consider implementing the trait `std::iter::Iterator` or choosing a less ambiguous method name
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#should_implement_trait

```